discord/commands/options.py

```diff
@@ -23,13 +23,40 @@
 """
 
 import inspect
-from typing import Any, Dict, List, Literal, Optional, Union
+from typing import TYPE_CHECKING, Dict, List, Literal, Optional, Type, Union
 from enum import Enum
 
-from ..abc import GuildChannel
+from ..abc import GuildChannel, Mentionable
 from ..channel import TextChannel, VoiceChannel, StageChannel, CategoryChannel, Thread
 from ..enums import ChannelType, SlashCommandOptionType, Enum as DiscordEnum
 
+if TYPE_CHECKING:
+    from ..ext.commands import Converter
+    from ..user import User
+    from ..member import Member
+    from ..message import Attachment
+    from ..role import Role
+
+    InputType = Union[
+        Type[str],
+        Type[bool],
+        Type[int],
+        Type[float],
+        Type[GuildChannel],
+        Type[Thread],
+        Type["ThreadOption"],
```

Comment:
Why does both exist in the first place?

Reply:
The ThreadOption class allows the dev to set which thread types can be selected I saw line 201 checks if ThreadOption is passed here so I decided to add it. We can remove it from both places.

Reply:
Ah okay

Reply:
Actually I think it doesn't make any sense to pass a ThreadOption inside an Option so I'll just remove that.